tools/sgartifactregistry/auth.go

```diff
@@ -0,0 +1,53 @@
+package sgartifactregistry
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"strings"
+
+	"go.einride.tech/sage/sg"
+	"go.einride.tech/sage/tools/sggcloud"
+)
+
+// Develop starts the Cloud Run service at the provided Go path with the provided service account and config.
+func NPMAuthenticate(ctx context.Context, registryURL string) error {
+	// Grab Google Cloud Access Token
+	var accessTokenOutput strings.Builder
+	cmd := sggcloud.Command(ctx, "auth", "print-access-token")
+	cmd.Stdout = &accessTokenOutput
+	if err := cmd.Run(); err != nil {
+		return err
+	}
+	registry := strings.TrimPrefix(registryURL, "https://")
+	// Set npm config
+	cmd = sg.Command(
+		ctx,
+		"npm",
+		"config",
+		"set",
+		"-L",
+		"user",
+		fmt.Sprintf("//%s/:_authToken", registry),
+		strings.TrimSpace(accessTokenOutput.String()),
+	)
+	// Discard stdout to avoid tokens being printed
+	cmd.Stdout = io.Discard
+	if err := cmd.Run(); err != nil {
+		return err
+	}
+	// Set yarn config
+	cmd = sg.Command(
+		ctx,
+		"yarn",
+		"config",
+		"set",
+		"--home",
+		"npmRegistries[\"//europe-west1-npm.pkg.dev/e-npm-registry-prod/npm\"].npmAuthToken",
```

Comment:
Shouldn't this also use `registry`?

Reply:
Oops that was a copy/paste error. Fixed it